### PR TITLE
Key pair generation

### PR DESCRIPTION
**Pull request by dbejanishvili**
key-pair storage (private key stored in LocalStorage, public key stored on the server)

### CODE REVIEW

File: webapp/src/api_client/api_client.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//store public key on server
+export function storePublicKey(publicKey, callback) {
```
Comment (bakurits) [inline webapp/src/api_client/api_client.js]: I think it would be better to implement these methods and then merge

File: server/API.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+func getUserIDFromRequest(r *http.Request) (string, error) {
+	userID := r.Header.Get(UserIDHeaderKey)
+	if userID == "" {
+		return "", errors.New("not authorized")
+	}
+	return userID, nil
+}
+
+// APICallHandler api call handler interface
+type APICallHandler func(p *Plugin, c *plugin.Context, w http.ResponseWriter, r *http.Request)
```
Comment (bakurits) [inline server/API.go]: Why are those functions which have `*Plugin` as a parameter and not methods of `Plugin`?

**Comment by iomodo** [comment]
Why was this PR closed?

**Comment by bakurits** [comment]
@iomodo  we decided to refactor this code with Golang best practices. It will be done in a few days and then we will reopen it

**Pull request closed without merging**